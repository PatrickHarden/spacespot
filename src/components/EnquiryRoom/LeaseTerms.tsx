import React, { useEffect, useState } from 'react'
import { useIntl } from 'react-intl'
import { useSelector, useDispatch } from 'react-redux'
import styled from 'styled-components'
import moment from 'moment'
import { get } from 'lodash'

import device from 'services/global/device'
import selectors from 'services/negotiation/selectors'
import actions from 'services/negotiation/actions'
import { getNegotiationTermArray } from 'services/negotiation/helpers'
import {
  useNegotiationFieldsIds,
  useNegotiationTerms,
} from 'services/negotiation/hooks'
import {
  NegotiationTerms,
  TermStatus,
  DepositType,
  PaymentType,
} from 'services/negotiation/types'
import enquirySelectors from 'services/enquiry/selectors'
import userSelectors from 'services/user/selectors'
import LeaseTermView from './LeaseTermView'
import LeaseTermForm, { FormErrors } from './LeaseTermForm'

import OutlinedButton from 'components/common/OutlinedButton'
import FilledButton from 'components/common/FilledButton'
import Spinner from 'components/icons/Spinner'

const ModButton = styled(OutlinedButton)`
  box-sizing: border-box;
  min-width: 110px;
`

const CancelButton = styled(OutlinedButton)`
  color: #6cb9d5;
  background-color: #ffffff;
  box-sizing: border-box;
  min-width: 110px;
  float: right;
  margin-right: 20px;
  @media ${device.mobile} {
    margin-right: 5px;
  }
`

const SubmitButton = styled(FilledButton)`
  color: white;
  background-color: #6cb9d5;
  box-sizing: border-box;
  min-width: 110px;
  float: right;
`

const ChangeButton = styled.button`
  padding-top: 0;
  color: #6d6c6c;
  border: none;
  outline: none;
  background: white;
  font-size: 16px;
  font-weight: 600;
  text-transform: uppercase;
  cursor: pointer;
`

const ModActions = styled.div`
  margin-top: 10px;
  display: flex;
  justify-content: space-between;
`
export const getLeaseStatusColor = (status: TermStatus) =>
  get(
    {
      [TermStatus.Declined]: '#DB4437',
      [TermStatus.Proposed]: '#EBBF59',
      [TermStatus.Accepted]: '#6CB9D5',
    },
    status,
    '#CECECE',
  )

const StatusText = styled.span<{ status: TermStatus }>`
  color: ${props => getLeaseStatusColor(props.status)};
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
  text-align: right;
`

const TermsHeading = styled.h2`
  font-size: 18px;
  font-weight: 500;
  line-height: 23px;
  margin-bottom: 5px;
`

const Container = styled.div`
  @media ${device.mobile} {
    margin: 0;
  }
`

const TermActions = styled.div`
  margin-top: 20px;
  min-height: 50px;
`

const RejectedBox = styled.div`
  padding: 5px;
  margin-bottom: 15px;
  border: 1px solid #ebbf59;
  background-color: rgba(235, 191, 89, 0.3);
  font-weight: 500;
`

const TextAccepting = styled.div`
  color: #6d6c6c;
  font-size: 14px;
  font-weight: 400;
  text-align: right;
  float: right;
  padding-top: 10px;
  width: 270px;
`

const LeaseTerms = (props: { disabled: boolean }) => {
  const { disabled } = props
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const dispatch = useDispatch()
  const [edit, setEdit] = useState(false)
  const isLandlord = useSelector(userSelectors.isLandlord)
  const enquiry = useSelector(enquirySelectors.getEnquiry)
  const enquiryStatus = get(enquiry, 'enquireStatus')
  const landlordId = get(enquiry, 'peer2Id')
  const status = useSelector(selectors.getStatus)

  const initialTerms = useNegotiationTerms()
  if (initialTerms) {
    if (!initialTerms.depositType) {
      initialTerms.depositType = DepositType.Cash
    }
    if (!initialTerms.frequency) {
      initialTerms.frequency = PaymentType.Monthly
    }
  }
  const fieldIds = useNegotiationFieldsIds()
  const [terms, setTerms] = useState(initialTerms)
  const [formErrors, setErrors] = useState({} as FormErrors)

  useEffect(() => {
    if (initialTerms) {
      setTerms(initialTerms)
    }
  }, [initialTerms, setTerms])

  useEffect(() => {
    if (isLandlord && status === TermStatus.Created) {
      setEdit(true)
    }
  }, [isLandlord, status])

  const requiredNum = (x: any) => {
    if (isNaN(x) || x === '') return t('VAL_REQUIRED')
    if (x < 0) return t('VAL_INVALID')
    return undefined
  }

  const valDate = (x: any) => {
    const d = moment(x)
    if (!d.isValid()) return t('VAL_NULL')
    return undefined
  }

  const validate = (terms: NegotiationTerms) => {
    const errors: FormErrors = {}
    if (!enquiry.flex) {
      errors.area = requiredNum(terms.area)
      errors.deposit = requiredNum(terms.deposit)
      errors.serviceCharges = requiredNum(terms.serviceCharges)
    } else {
      errors.noOfDesks = requiredNum(terms.noOfDesks)
    }
    errors.duration = requiredNum(terms.duration)
    errors.rent = requiredNum(terms.rent)
    errors.start = valDate(terms.start)

    // remove undefined fields
    const acc: FormErrors = Object.keys(errors).reduce((acc, key) => {
      if (errors[key] !== undefined) acc[key] = errors[key]
      return acc
    }, {} as FormErrors)
    return acc
  }

  const update = (val: NegotiationTerms) => {
    const errors = validate(val)
    setErrors(errors)
    setTerms(val)
  }

  const onCancel = () => {
    setEdit(false)
    setTerms(initialTerms)
  }
  const onAccept = () => {
    dispatch(actions.accept())
  }
  const onDecline = () => {
    dispatch(actions.reject())
  }
  const onSubmit = () => {
    const errors = validate(terms)
    if (Object.keys(errors).length > 0) {
      setErrors(errors)
    } else {
      const values = getNegotiationTermArray(terms, fieldIds)
      dispatch(actions.putCounter(values))
      setEdit(false)
    }
  }

  if (
    enquiryStatus === TermStatus.Pending ||
    (!isLandlord && status === TermStatus.Created)
  ) {
    return (
      <Container>
        <p>{t('NEGOTIATE_EXPLANATION_LEASE_TERMS')}</p>
      </Container>
    )
  }

  if (enquiry === undefined) {
    return (
      <Container>
        <Spinner />
      </Container>
    )
  }

  return (
    <Container>
      {status !== TermStatus.Accepted && status !== TermStatus.Declined && (
        <TermsHeading>
          {isLandlord
            ? t('NEGOTIATE_YOUR_TERMS_LANDLORD')
            : t('NEGOTIATE_YOUR_TERMS')}
        </TermsHeading>
      )}
      {status === TermStatus.Declined && (
        <RejectedBox>
          {isLandlord
            ? t('NEGOTIATE_REJECT_MSG')
            : t('NEGOTIATE_TENANT_REJECT_MSG')}
        </RejectedBox>
      )}
      {!disabled && edit ? (
        <LeaseTermForm
          terms={terms}
          setTerms={update}
          errors={formErrors}
          isFlex={enquiry.flex}
          flexType={enquiry.flexType}
        />
      ) : (
        <LeaseTermView
          status={status}
          terms={terms}
          userId={landlordId}
          isFlex={enquiry.flex}
          flexType={enquiry.flexType}
          disabled={disabled}
        />
      )}
      {disabled ? null : isLandlord ? (
        <>
          {edit ? (
            <TermActions>
              <SubmitButton data-testid="submit" onClick={onSubmit}>
                {t('NEGOTIATE_SUBMIT')}
              </SubmitButton>
              <CancelButton data-testid="cancel" onClick={onCancel}>
                {t('NEGOTIATE_CANCEL')}
              </CancelButton>
            </TermActions>
          ) : (
            <ModActions>
              <div>
                <ModButton
                  data-testid="modify"
                  onClick={() => {
                    setEdit(true)
                  }}>
                  {t('NEGOTIATE_MODIFY')}
                </ModButton>
              </div>
              <div>
                <StatusText status={status}>
                  {get(
                    {
                      [TermStatus.Accepted]: t('NEGOTIATE_STATUS_ACCEPTED'),
                      [TermStatus.Declined]: t('NEGOTIATE_STATUS_REJECTED'),
                      [TermStatus.Proposed]: t(
                        'NEGOTIATE_STATUS_WAITING_CONFIRMATION',
                      ),
                    },
                    status,
                  )}
                </StatusText>
              </div>
            </ModActions>
          )}
        </>
      ) : status === TermStatus.Accepted || status === TermStatus.Declined ? (
        <ModActions>
          <div>
            {status === TermStatus.Accepted ? (
              <ChangeButton data-testid="decline" onClick={onDecline}>
                {t('NEGOTIATE_DECLINE')}
              </ChangeButton>
            ) : (
              <ChangeButton data-testid="accept" onClick={onAccept}>
                {t('NEGOTIATE_ACCEPT')}
              </ChangeButton>
            )}
          </div>
          <div>
            <StatusText status={status}>
              {t(
                status === TermStatus.Accepted
                  ? 'NEGOTIATE_STATUS_ACCEPTED'
                  : 'NEGOTIATE_STATUS_REJECTED',
              )}
            </StatusText>
          </div>
        </ModActions>
      ) : (
        <TermActions>
          <SubmitButton data-testid="accept" onClick={onAccept}>
            {t('NEGOTIATE_ACCEPT')}
          </SubmitButton>
          <CancelButton data-testid="decline" onClick={onDecline}>
            {t('NEGOTIATE_DECLINE')}
          </CancelButton>
        </TermActions>
      )}
      {status === TermStatus.Proposed || status === TermStatus.Declined ? (
        <TextAccepting>
          {isLandlord ? null : t('NEGOTIATE_SPECIAL_PROVISION_ACCEPTING')}
        </TextAccepting>
      ) : null}
    </Container>
  )
}

export default LeaseTerms
