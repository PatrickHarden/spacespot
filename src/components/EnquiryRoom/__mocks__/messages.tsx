export const messages = [
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message',
    sentAt: '2018-10-18T12:40:35.886+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message1',
    sentAt: '2018-10-18T12:40:43.547+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message2',
    sentAt: '2018-10-18T12:40:44.946+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message3',
    sentAt: '2018-10-18T12:40:45.678+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message4',
    sentAt: '2018-10-18T12:40:46.856+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message5',
    sentAt: '2018-10-23T14:17:59.662+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'jk',
    sentAt: '2018-10-28T10:13:59.805+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message6',
    sentAt: '2018-10-28T10:16:16.585+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '11:44',
    sentAt: '2018-10-28T10:45:02.592+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:34',
    sentAt: '2018-10-28T11:23:27.467+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'll',
    sentAt: '2018-10-28T11:29:37.057+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:30',
    sentAt: '2018-10-28T11:30:13.643+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'The message',
    sentAt: '2018-10-29T08:51:21.007+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Another',
    sentAt: '2018-10-29T09:28:49.589+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Other',
    sentAt: '2018-10-29T09:29:02.900+0000',
  },
]
