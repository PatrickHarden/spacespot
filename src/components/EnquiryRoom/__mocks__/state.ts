export const mockState: any = {
  enquiry: {
    data: {
      spaceId: 'SS_SPACE_665',
      peer1Id: '1',
      peer2Id: '2',
      chatId: 159,
      enquireStatus: 'ACCEPTED',
      negotiationId: 139,
      enquireId: 159,
      flex: false,
    },
  },
  negotiation: {
    data: {
      flexible: false,
      terms: [
        {
          termId: 5,
          value: 'MONTHLY',
        },
        {
          termId: 6,
          value: '2000',
        },
        {
          termId: 7,
          value: '2',
        },
        {
          termId: 8,
          value: '2000',
        },
        {
          termId: 9,
          value: 'GUARANTEE',
        },
        {
          termId: 1,
          value: '3500',
        },
        {
          termId: 2,
          value: '40000',
        },
        {
          termId: 3,
          value: '2020-04-22',
        },
        {
          termId: 4,
          value: '13',
        },
      ],
      negotiationId: 139,
      enquireId: 159,
      createdAt: '2020-02-14T06:30:08.354+0000',
      lastUpdated: '2020-03-05T16:35:02.513+0000',
      lastModifyBy: '0062fb00-e389-4bb8-9119-613e3c45938a',
      status: 'PROPOSED',
      customFitout: false,
      fitoutOption: 0,
      specialProvision: 'gfd',
      specialProvisionStatus: 'CREATED',
    },
    loading: false,
    termsFields: {
      '1': 'Area',
      '2': 'Rent',
      '3': 'Start',
      '4': 'Duration',
      '5': 'Frequency',
      '6': 'AdditionalCosts',
      '7': 'Deposit',
      '8': 'ServiceCharges',
      '9': 'DepositType',
    },
    signerInfo: {
      id: 1,
      name: 'Name1',
      emailId: 'name1@email.com',
      companyName: 'companyA',
      companyNumber: '12345678',
    },
    customFitout: [],
  },
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
