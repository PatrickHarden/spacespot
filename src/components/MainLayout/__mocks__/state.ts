export const mockState: any = {
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
