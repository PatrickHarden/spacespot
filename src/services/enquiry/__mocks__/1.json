{
  "chatId": 1,
  "enquireId": 1,
  "spaceId": 2,
  "peer1Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda3",
  "peer2Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda4"
}
