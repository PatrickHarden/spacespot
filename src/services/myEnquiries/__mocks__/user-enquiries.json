[
  {
    "enquireId": 167,
    "negotiationId": 147,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 17:59:47 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "123" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T16:59:53.122+0000"
      }
    ]
  },
  {
    "enquireId": 168,
    "negotiationId": 148,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 18:00:28 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "13" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T17:00:34.014+0000"
      }
    ]
  }
]
