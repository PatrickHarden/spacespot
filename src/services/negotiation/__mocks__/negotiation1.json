{
  "terms": [
    { "termId": 1, "value": "34" },
    { "termId": 2, "value": "207" },
    { "termId": 3, "value": "29 Months" },
    { "termId": 4, "value": "200 Sq Mtr" },
    { "termId": 5, "value": "$5600 per month" },
    { "termId": 6, "value": "20th September" }
  ],
  "negotiationId": 129,
  "enquireId": 149,
  "createdAt": "2020-02-03T13:03:07.661+0000",
  "lastUpdated": "2020-02-11T09:06:15.764+0000",
  "lastModifyBy": "c700793a-1963-4b2b-996f-1939f853cc10",
  "status": "CREATED"
}
