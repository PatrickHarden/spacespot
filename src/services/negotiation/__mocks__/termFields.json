{
  "1": "Area",
  "2": "Rent",
  "3": "Start",
  "4": "Duration",
  "5": "Frequency",
  "6": "AdditionalCosts",
  "7": "Deposit",
  "8": "ServiceCharges"
}
