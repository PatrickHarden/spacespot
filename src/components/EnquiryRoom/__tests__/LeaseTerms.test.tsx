import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'

import LeaseTerms from '../LeaseTerms'
import { mockState } from '../__mocks__/state'

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

it('Renders landlord view (PROPOSED status)', () => {
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <LeaseTerms disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByTestId('modify'))
  fireEvent.click(wrapper.getByTestId('cancel'))
  fireEvent.click(wrapper.getByTestId('modify'))
  fireEvent.change(wrapper.getByTestId('duration'), { target: { value: '10' } })
  fireEvent.change(wrapper.getByTestId('area'), { target: { value: '4000' } })
  fireEvent.change(wrapper.getByTestId('rent'), { target: { value: '4000' } })
  fireEvent.change(wrapper.getByTestId('serviceCharges'), {
    target: { value: '4000' },
  })
  fireEvent.click(wrapper.getByTestId('submit'))
})

it('Renders tenant view (PROPOSED status)', () => {
  // eslint-disable-next-line @typescript-eslint/camelcase
  mockState.user.auth.idToken.claims.extension_Landlord = false
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <LeaseTerms disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByTestId('accept'))
})

it('Renders tenant view (ACCEPTED status)', () => {
  // eslint-disable-next-line @typescript-eslint/camelcase
  mockState.user.auth.idToken.claims.extension_Landlord = false
  mockState.negotiation.data.status = 'ACCEPTED'
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <LeaseTerms disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByTestId('decline'))
})
