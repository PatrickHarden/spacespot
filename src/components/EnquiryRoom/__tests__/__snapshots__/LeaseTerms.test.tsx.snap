// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders landlord view (PROPOSED status) 1`] = `
<DocumentFragment>
  <div
    class="sc-bRBYWo gFRiVi"
  >
    <h2
      class="sc-Rmtcm krzlQq"
    >
      NEGOTIATE_YOUR_TERMS_LANDLORD
    </h2>
    <div
      class="sc-EHOje jZQcXK"
    >
      <svg
        class="sc-gqjmRU cugBST"
        height="54"
        viewBox="-2 -2 58 58"
        width="54"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="white"
          fill-rule="evenodd"
          opacity="1"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#ffcdc8"
            fill-opacity="1"
            fill-rule="nonzero"
            stroke="#fe7062"
            stroke-width="2"
          >
            <circle
              cx="27"
              cy="27"
              r="27"
            />
          </g>
          <g
            fill="#fe7062"
            fill-rule="nonzero"
          >
            <text
              font-size="26"
              font-weight="400"
              id="name"
              text-anchor="middle"
            >
              <tspan
                x="27"
                y="37"
              >
                IH
              </tspan>
            </text>
          </g>
        </g>
      </svg>
      <div
        class="sc-bZQynM hnnyjN"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_AREA
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          3500sq.m.
        </div>
      </div>
      <div
        class="sc-gzVnrw cZMWQi"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_LENGTH
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          13 NEGOTIATE_MONTHS NEGOTIATE_FROM ONBOARDING_AVAILABILITY_NOW
        </div>
      </div>
      <div
        class="sc-htoDjs hXNBKs"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_MONTHLY NEGOTIATE_PAYMENT
        </h2>
        <div
          class="sc-iwsKbI hiqfNw"
        >
          <div
            class="sc-gZMcBi fYhffy"
          >
            <div
              class="sc-ifAKCX dXjDeY"
            >
              40000kr
            </div>
            <div
              class="sc-bxivhb etrvNB"
            >
              NEGOTIATE_RENT
            </div>
          </div>
          <div
            class="sc-gZMcBi fYhffy"
          >
            <div
              class="sc-ifAKCX dXjDeY"
            >
              2000kr
            </div>
            <div
              class="sc-bxivhb etrvNB"
            >
              NEGOTIATE_SERVICE_CHARGES
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-dnqmqq gJJfia"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_DEPOSIT
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          2 NEGOTIATE_DEPOSIT_IN NEGOTIATE_DEPOSIT_GARANTEE
        </div>
      </div>
    </div>
    <div
      class="sc-gipzik bJCGzT"
    >
      <div>
        <button
          class="sc-brqgnP sc-jDwBTQ jokWaa"
          data-testid="modify"
        >
          NEGOTIATE_MODIFY
        </button>
      </div>
      <div>
        <span
          class="sc-csuQGl dcYdvi"
        >
          NEGOTIATE_STATUS_WAITING_CONFIRMATION
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Renders tenant view (ACCEPTED status) 1`] = `
<DocumentFragment>
  <div
    class="sc-bRBYWo gFRiVi"
  >
    <div
      class="sc-EHOje jXHpcx"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-bwzfXH eHKxnV"
        color="#6CB9D5"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <svg
        class="sc-gqjmRU cugBST"
        height="54"
        viewBox="-2 -2 58 58"
        width="54"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="white"
          fill-rule="evenodd"
          opacity="1"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#ffcdc8"
            fill-opacity="1"
            fill-rule="nonzero"
            stroke="#fe7062"
            stroke-width="2"
          >
            <circle
              cx="27"
              cy="27"
              r="27"
            />
          </g>
          <g
            fill="#fe7062"
            fill-rule="nonzero"
          >
            <text
              font-size="26"
              font-weight="400"
              id="name"
              text-anchor="middle"
            >
              <tspan
                x="27"
                y="37"
              >
                IH
              </tspan>
            </text>
          </g>
        </g>
      </svg>
      <div
        class="sc-bZQynM hnnyjN"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_AREA
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          3500sq.m.
        </div>
      </div>
      <div
        class="sc-gzVnrw cZMWQi"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_LENGTH
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          13 NEGOTIATE_MONTHS NEGOTIATE_FROM ONBOARDING_AVAILABILITY_NOW
        </div>
      </div>
      <div
        class="sc-htoDjs hXNBKs"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_MONTHLY NEGOTIATE_PAYMENT
        </h2>
        <div
          class="sc-iwsKbI hiqfNw"
        >
          <div
            class="sc-gZMcBi fYhffy"
          >
            <div
              class="sc-ifAKCX dXjDeY"
            >
              40000kr
            </div>
            <div
              class="sc-bxivhb etrvNB"
            >
              NEGOTIATE_RENT
            </div>
          </div>
          <div
            class="sc-gZMcBi fYhffy"
          >
            <div
              class="sc-ifAKCX dXjDeY"
            >
              2000kr
            </div>
            <div
              class="sc-bxivhb etrvNB"
            >
              NEGOTIATE_SERVICE_CHARGES
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-dnqmqq gJJfia"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_DEPOSIT
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          2 NEGOTIATE_DEPOSIT_IN NEGOTIATE_DEPOSIT_GARANTEE
        </div>
      </div>
    </div>
    <div
      class="sc-gipzik bJCGzT"
    >
      <div>
        <button
          class="sc-jlyJG humxlj"
          data-testid="decline"
        >
          NEGOTIATE_DECLINE
        </button>
      </div>
      <div>
        <span
          class="sc-csuQGl iyLCjV"
        >
          NEGOTIATE_STATUS_ACCEPTED
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Renders tenant view (PROPOSED status) 1`] = `
<DocumentFragment>
  <div
    class="sc-bRBYWo gFRiVi"
  >
    <h2
      class="sc-Rmtcm krzlQq"
    >
      NEGOTIATE_YOUR_TERMS
    </h2>
    <div
      class="sc-EHOje jZQcXK"
    >
      <svg
        class="sc-gqjmRU cugBST"
        height="54"
        viewBox="-2 -2 58 58"
        width="54"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="white"
          fill-rule="evenodd"
          opacity="1"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#ffcdc8"
            fill-opacity="1"
            fill-rule="nonzero"
            stroke="#fe7062"
            stroke-width="2"
          >
            <circle
              cx="27"
              cy="27"
              r="27"
            />
          </g>
          <g
            fill="#fe7062"
            fill-rule="nonzero"
          >
            <text
              font-size="26"
              font-weight="400"
              id="name"
              text-anchor="middle"
            >
              <tspan
                x="27"
                y="37"
              >
                IH
              </tspan>
            </text>
          </g>
        </g>
      </svg>
      <div
        class="sc-bZQynM hnnyjN"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_AREA
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          3500sq.m.
        </div>
      </div>
      <div
        class="sc-gzVnrw cZMWQi"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_LENGTH
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          13 NEGOTIATE_MONTHS NEGOTIATE_FROM ONBOARDING_AVAILABILITY_NOW
        </div>
      </div>
      <div
        class="sc-htoDjs hXNBKs"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_MONTHLY NEGOTIATE_PAYMENT
        </h2>
        <div
          class="sc-iwsKbI hiqfNw"
        >
          <div
            class="sc-gZMcBi fYhffy"
          >
            <div
              class="sc-ifAKCX dXjDeY"
            >
              40000kr
            </div>
            <div
              class="sc-bxivhb etrvNB"
            >
              NEGOTIATE_RENT
            </div>
          </div>
          <div
            class="sc-gZMcBi fYhffy"
          >
            <div
              class="sc-ifAKCX dXjDeY"
            >
              2000kr
            </div>
            <div
              class="sc-bxivhb etrvNB"
            >
              NEGOTIATE_SERVICE_CHARGES
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-dnqmqq gJJfia"
      >
        <h2
          class="sc-htpNat hDHSmr"
        >
          NEGOTIATE_DEPOSIT
        </h2>
        <div
          class="sc-ifAKCX dXjDeY"
        >
          2 NEGOTIATE_DEPOSIT_IN NEGOTIATE_DEPOSIT_GARANTEE
        </div>
      </div>
    </div>
    <div
      class="sc-hzDkRC yEbsH"
    >
      <button
        class="sc-cMljjf sc-iRbamj gfnDYA"
        data-testid="accept"
      >
        NEGOTIATE_ACCEPT
      </button>
      <button
        class="sc-brqgnP sc-gPEVay ehhnkY"
        data-testid="decline"
      >
        NEGOTIATE_DECLINE
      </button>
    </div>
  </div>
</DocumentFragment>
`;
